source/developers/sts-for-operator.rst

```diff
@@ -0,0 +1,132 @@
+.. _minio-sts-operator:
+
+===============================================
+Security Token Service (STS) for MinIO Operator
+===============================================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+.. versionadded:: Operator v5.0.0
+
+   The MinIO Operator supports a set of API calls that allows an application to obtain STS credentials for a MinIO Tenant.
+
+:ref:`STS credentials <minio-security-token-service>` allow an application to access objects on a MinIO Tenant without the need to create any credentials for the application on the tenant.
+
+.. important:: 
+
+   MinIO Operator launches with STS *disabled* by default.
+   To use STS with the Operator, you must first explicitly enable it.
+
+How STS Authorization Works in Kubernetes
+-----------------------------------------
+
+An application can use an ``AssumeRoleWithWebIdentity`` call to send a request for temporary credentials to the MinIO Operator.
+The Operator checks the validity of the request, retrieves policies for the application, obtains credentials from the tenant, and then passes the credentials back the application.
+The application uses the issued credentials to work with the object storage on the tenant.
+
+The complete process includes the following steps:
+
+1. An application sends an API request to the MinIO Operator containing the name of the application and a service account to use.
+2. The MinIO Operator uses the Kubernetes API to check that the JWT is a valid request.
```

Comment:
Is there a "first reference" expansion for `JWT` nearby?

Reply:
good point @feorlen , JWT is a JSON web Token, but could expand the description of where it comes from, somenthing like this:


The STS API expects a JSON Web Token (JWT), it is a `Kubernetes service Account Token <https://kubernetes.io/docs/reference/access-authn-authz/authentication/#service-account-tokens>`.

When a pod have a Kubernetes service account associated to it, it will have the JWT  mounted in a well known location in the filesystem, that i  "/var/run/secrets/kubernetes.io/serviceaccount/token".

Take for example the following kubernetes manifest:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  namespace: sts-client
  name: my-service-account
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  namespace: default
spec:
  replicas: 3
  template:
    metadata:
    # ...
    spec:
      serviceAccountName: my-service-account
      containers:
      - name: nginx
        image: nginx:1.14.2

```
In the first yaml document of the manifest a service account `my-service-account` is created.
In the second yaml document a Deployment is created, and is associated to the service account `my-service-account` in the `spec.spec.serviceAccountName` field.

Reply:
@pjuarezd ooo nice extended description. A bit long to fit in this list, but maybe we can find somewhere useful. @djwfyi ? 

Thanks!

Reply:
Thanks for this, @pjuarezd.
I expanded the first instance of JWT, then included portions of this description in several places.
For example, part of it in the last step of the procedure where the SDK call happens. And more of it in the Service Account description under Example Resources.